### PR TITLE
157544932 add messaging configuration

### PR DESCRIPTION
**Pull request by KyleTTownsend**

### CODE REVIEW

**Comment by Hoagithor** [comment]
Note UserController constructors in existing tests were given null instead of an IQueueFactory since they do not call queue methods. Future tests will require a valid an IQueueFactory if they test the queue methods

File: src/ServiceHub.User.Service/Controllers/UserController.cs [hunk 1]
```diff
@@ -6,19 +6,27 @@
...
 {
     [Route("api/[controller]")]
     public class UserController : BaseController
     {
         private readonly IUserRepository _userRepository;
+        private readonly IQueueFactory _queues;
 
         public UserController(IUserRepository userRepository,
-                              ILoggerFactory loggerFactory)
+                              ILoggerFactory loggerFactory, IQueueFactory queues)
```
Comment (escalonn) [inline src/ServiceHub.User.Service/Controllers/UserController.cs]: put this guy on his own line

File: src/ServiceHub.User.Service/Controllers/UserController.cs [hunk 2]
```diff
@@ -6,19 +6,27 @@
...
           : base(loggerFactory)
         {
+            
             _userRepository = userRepository;
+            _queues = queues;
+
```
Comment (escalonn) [inline src/ServiceHub.User.Service/Controllers/UserController.cs]: unnecessary whitespace

File: src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+using Microsoft.Azure.ServiceBus;
+namespace ServiceHub.User.Service.ServiceBus
+{
+    public class QueueFactory : IQueueFactory
+    {
+        private string connectionString;
+        public QueueFactory(string connection)
+        {
+            connectionString = connection;
+
```
Comment (escalonn) [inline src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs]: blank line between methods, but not at the end of a method

File: src/ServiceHub.User.Service/Startup.cs [hunk 4]
```diff
@@ -25,13 +26,16 @@ public void ConfigureServices(IServiceCollection services)
...
+            
             services.AddSingleton<IMongoCollection<Context.Models.User>>(mongoCollection =>
             {
                 return new MongoClient(connectionString)
                 .GetDatabase("userdb")
                 .GetCollection<Context.Models.User>("users");
             });
+            string busConnection = Configuration.GetConnectionString("ServiceBus");
+
+            services.AddScoped<IQueueFactory>(sp => new QueueFactory(busConnection));
```
Comment (escalonn) [inline src/ServiceHub.User.Service/Startup.cs]: this can be a singleton

File: src/ServiceHub.User.Testing/Service/UserControllerTests.cs [hunk 5]
```diff
@@ -125,7 +125,7 @@ public async Task ValidGetById_Return200_And_Model()
             var mockRepo = new Mock<IUserRepository>();
             mockRepo.Setup(x => x.GetByIdAsync(It.IsAny<Guid>())).ReturnsAsync(((Guid y) => contextUsers.First(z => z.UserId == y)));
 
-            UserController c = new UserController(mockRepo.Object, new LoggerFactory());
+            UserController c = new UserController(mockRepo.Object, new LoggerFactory(),null);
```
Comment (escalonn) [inline src/ServiceHub.User.Testing/Service/UserControllerTests.cs]: spacing

File: src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus;
+namespace ServiceHub.User.Service.ServiceBus
+{
+    public class QueueFactory : IQueueFactory
+    {
+        private string connectionString;
```
Comment (fredbelotte) [inline src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs]: Code Smell Code Smell: Make 'connectionString' 'readonly'. ([csharpsquid:S2933](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS2933))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Auser&pullRequest=91&issues=AWVYYz2fiZQY_aXE9PZl&open=AWVYYz2fiZQY_aXE9PZl)

[ISSUE_KEY]: AWVYYz2fiZQY_aXE9PZl

File: src/ServiceHub.User.Service/Controllers/UserController.cs [hunk 7]
```diff
@@ -6,19 +6,28 @@
...
+using System.Text;
+using ServiceHub.User.Service.ServiceBus;
 
 namespace ServiceHub.User.Service.Controllers
 {
     [Route("api/[controller]")]
     public class UserController : BaseController
     {
         private readonly IUserRepository _userRepository;
+        private readonly IQueueFactory _queues;
```
Comment (fredbelotte) [inline src/ServiceHub.User.Service/Controllers/UserController.cs]: Code Smell Code Smell: Remove the '_queues' field and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Auser&pullRequest=91&issues=AWVYYz2XiZQY_aXE9PZk&open=AWVYYz2XiZQY_aXE9PZk)

[ISSUE_KEY]: AWVYYz2XiZQY_aXE9PZk

Comment (fredbelotte) [inline src/ServiceHub.User.Service/Controllers/UserController.cs]: Code Smell Code Smell: Remove the '_queues' field and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Auser&pullRequest=91&issues=AWVYYz2XiZQY_aXE9PZk&open=AWVYYz2XiZQY_aXE9PZk)

[ISSUE_KEY]: AWVYYz2XiZQY_aXE9PZk

**Comment by escalonn** [comment]
looks like there is an outstanding code smell, and also, we need to merge from broker

**Comment by Hoagithor** [comment]
Merged from broker. Should have mentioned: code-smell is related to setting up a QueueFactory to use for implementation step, and unavoidable if we desire complete configuration for this story.
